### PR TITLE
Api 36674 new supp claim form

### PR DESCRIPTION
**Pull request by michelpmcdonald**
## Summary

New Supplemental Claim form, we are calling it v4.

New form is behind flipper flag,  decision_review_sc_form_v4_enabled.  We don't have any immediate plans to take this live at this point.

## Related issue(s)

API-36674

## Testing done

- [x] *New code is covered by unit tests*

We have standard test data for supp claims.  Used test data to generate forms and manually confirmed the forms are as expected.  The generated files become PDF reference unit test fixtures, unit test generates forms and the compairs the generated from with the ref fixture.

## What areas of the site does it impact?
Intake will start receiving and processing the latest version of the Va supp claim form.
Since the form we generate is an internal implementation detail and not directly exposed to consumers, there are no changes for them at this point.

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [x]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x]  Feature/bug has a monitor built into Datadog (if applicable)

### CODE REVIEW

File: modules/appeals_api/spec/requests/appeals_api/v2/decision_reviews/supplemental_claims_spec.rb [hunk 1]
```diff
@@ -130,6 +130,7 @@ def make_request(headers)
         post(path, params: extra_data, headers: headers.merge({ 'X-VA-ICN' => icn }))
 
         expect(response).to have_http_status(:unprocessable_entity)
+        puts response.body
```
[the remark this answers is not in the document]
Comment (michelpmcdonald) [reply modules/appeals_api/spec/requests/appeals_api/v2/decision_reviews/supplemental_claims_spec.rb:133]: temp debugging stuff, i left that in for YOU :)

File: modules/appeals_api/lib/appeals_api/sc_evidence.rb [hunk 2]
```diff
@@ -6,16 +6,20 @@ def location
...
       evidence_dates = attributes['evidenceDates']
 
       return [''] if evidence_dates.blank?
 
       evidence_dates.map do |hash|
-        if hash['startDate'] == hash['endDate']
+        if (hash['startDate'] == hash['endDate']) && !month_format
           hash['startDate']
-        else
+        elsif !month_format
```
[the remark this answers is not in the document]
Comment (michelpmcdonald) [reply modules/appeals_api/lib/appeals_api/sc_evidence.rb]: split them,

File: modules/appeals_api/spec/services/appeals_api/pdf_construction/generator_spec.rb [hunk 3]
```diff
@@ -219,6 +219,12 @@
           international_phone: '+WWW-WWWWWWWWWWWWWWW'
         }
       end
+
+      context 'v4' do
+        include_examples 'shared SC v2 and v3 generator examples', 'v4', {
```
[the remark this answers is not in the document]
Comment (michelpmcdonald) [reply modules/appeals_api/spec/services/appeals_api/pdf_construction/generator_spec.rb]: changed

File: modules/appeals_api/spec/support/shared_examples_for_gateway_origin_check.rb [hunk 4]
```diff
@@ -7,16 +7,18 @@ def make_request(_headers = {})
...
+
+        expect(response).to have_http_status(:ok)
+      end
     end
 
     context 'with benefits_require_gateway_origin flag on' do
-      before do
-        allow(Flipper).to receive(:enabled?).with(:benefits_require_gateway_origin).and_return(true)
-      end
+      before { Flipper.enable(:benefits_require_gateway_origin) }
```
[the remark this answers is not in the document]
Comment (michelpmcdonald) [reply modules/appeals_api/spec/support/shared_examples_for_gateway_origin_check.rb:21]: best single line of code ever written, who commited dat?

**Review by kristen-brown — APPROVED** [review]
Your changes look great! 🥇

**Review by rjohnson2011 — COMMENTED** [review]
https://github.com/department-of-veterans-affairs/vets-api/blob/fff739e5baaf6d45f0ede366f62915583691bc98/modules/appeals_api/spec/services/appeals_api/pdf_construction/generator_spec.rb#L47

Noticed you already did this for 118, do you mind updating so it's clear this is using shared v2/v3/v4 examples please?

**Review by rjohnson2011 — APPROVED** [review]

**Review by ericboehs — APPROVED** [review]

**Pull request merged**